salt/states/boto_cloudtrail.py

```diff
@@ -0,0 +1,353 @@
+# -*- coding: utf-8 -*-
+'''
+Manage CloudTrail Objects
+=================
+
+.. versionadded:: Boron
+
+Create and destroy CloudTrail objects. Be aware that this interacts with Amazon's services,
+and so may incur charges.
+
+This module uses ``boto3``, which can be installed via package, or pip.
+
+This module accepts explicit vpc credentials but can also utilize
+IAM roles assigned to the instance through Instance Profiles. Dynamic
+credentials are then automatically obtained from AWS API and no further
+configuration is necessary. More information available `here
+<http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html>`_.
+
+If IAM roles are not used you need to specify them either in a pillar file or
+in the minion's config file:
+
+.. code-block:: yaml
+
+    vpc.keyid: GKTADJGHEIQSXMKKRBJ08H
+    vpc.key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+
+It's also possible to specify ``key``, ``keyid`` and ``region`` via a profile,
+either passed in as a dict, or as a string to pull from pillars or minion
+config:
+
+.. code-block:: yaml
+
+    myprofile:
+        keyid: GKTADJGHEIQSXMKKRBJ08H
+        key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+            region: us-east-1
+
+.. code-block:: yaml
+
+    Ensure trail exists:
+        boto_cloudtrail.present:
+            - Name: mytrail
+            - S3BucketName: mybucket
+            - S3KeyPrefix: prefix
+            - region: us-east-1
+            - keyid: GKTADJGHEIQSXMKKRBJ08H
+            - key: askdjghsdfjkghWupUjasdflkdfklgjsdfjajkghs
+
+'''
+
+# Import Python Libs
+from __future__ import absolute_import
+import logging
+import os
+import os.path
+import json
+
+# Import Salt Libs
+import salt.utils
+from salt.ext.six import string_types
+
+log = logging.getLogger(__name__)
+
+
+def __virtual__():
+    '''
+    Only load if boto is available.
+    '''
+    return 'boto_cloudtrail' if 'boto_cloudtrail.exists' in __salt__ else False
+
+
+def present(name, Name,
+           S3BucketName, S3KeyPrefix=None,
+           SnsTopicName=None,
+           IncludeGlobalServiceEvents=True,
+           #IsMultiRegionTrail=None,
+           EnableLogFileValidation=False,
+           CloudWatchLogsLogGroupArn=None,
+           CloudWatchLogsRoleArn=None,
+           KmsKeyId=None,
+           LoggingEnabled=True,
+           Tags=None,
+           region=None, key=None, keyid=None, profile=None):
+    '''
+    Ensure trail exists.
+
+    name
+        The name of the state definition
+
+    Name
```

Comment:
This seems to break the paradigm that we generally have for `name` in states. Salt states will use the top-level key in a state as the name only if the `name` argument is not _explicitly_ provided as an argument.

What do you think about killing the `Name` argument here and using `name` instead? I think it might be less confusing.